QSB/TranslationSync/Events/SetAsTranslatedEvent.cs

```diff
@@ -0,0 +1,51 @@
+﻿using QSB.Events;
+using QSB.TranslationSync.WorldObjects;
+using QSB.WorldSync;
+
+namespace QSB.TranslationSync.Events
+{
+	public class SetAsTranslatedEvent : QSBEvent<SetAsTranslatedMessage>
+	{
+		public override EventType Type => EventType.TextTranslated;
+
+		public override void SetupListener() => GlobalMessenger<NomaiTextType, int, int>.AddListener(EventNames.QSBTextTranslated, Handler);
+		public override void CloseListener() => GlobalMessenger<NomaiTextType, int, int>.RemoveListener(EventNames.QSBTextTranslated, Handler);
+
+		private void Handler(NomaiTextType type, int objId, int textId) => SendEvent(CreateMessage(type, objId, textId));
+
+		private SetAsTranslatedMessage CreateMessage(NomaiTextType type, int objId, int textId) => new SetAsTranslatedMessage
+		{
+			AboutId = LocalPlayerId,
+			ObjectId = objId,
+			TextId = textId,
+			TextType = type
+		};
+
+		public override void OnReceiveRemote(bool server, SetAsTranslatedMessage message)
+		{
+			if (!QSBCore.HasWokenUp)
+			{
+				return;
+			}
+			if (message.TextType == NomaiTextType.WallText)
+			{
+				var obj = QSBWorldSync.GetWorldObject<QSBWallText>(message.ObjectId);
+				obj.HandleSetAsTranslated(message.TextId);
+			}
+			else if (message.TextType == NomaiTextType.Computer)
+			{
+				var obj = QSBWorldSync.GetWorldObject<QSBComputer>(message.ObjectId);
+				obj.HandleSetAsTranslated(message.TextId);
+			}
+			else if (message.TextType == NomaiTextType.VesselComputer)
+			{
+				var obj = QSBWorldSync.GetWorldObject<QSBVesselComputer>(message.ObjectId);
+				obj.HandleSetAsTranslated(message.TextId);
+			}
```

Comment:
if QSBWallText, QSBComputer and QSBVesselComputer inherits from common class/interface, you could extract some of this out of the if-else:

var obj = GetObj(...); // use if-else or switch
obj.HandleSetAsTranslated(message.TextId);

not a big deal tho, it's fine as-is ;)